### PR TITLE
Refactor back2top button

### PR DESCRIPTION
**Pull request by stevenjoezhang**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] Docs in [NexT website](https://theme-next.org/docs/) have been added / updated (for new features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [x] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue resolved: N/A

## What is the new behavior?

- Screenshots with this changes: N/A
- Link to demo site with this changes: N/A

### How to use?
In NexT `_config.yml`:
```yml
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by 1v9** [comment]
My b2t button lost, check [here](https://1v9.im/post/mermaid-support-in-next.html)

File: source/css/_common/components/components.styl [hunk 1]
```diff
@@ -5,7 +5,7 @@
 @import "pagination";
 @import "comments";
 @import "tag-cloud";
-@import hexo-config('sidebar.b2t') ? "back-to-top-sidebar" : "back-to-top";
+@import hexo-config('back2top.sidebar') ? "back-to-top-sidebar" : "back-to-top";
```
Comment (JiangTJ) [inline source/css/_common/components/components.styl]: Add `if hexo-config('back2top.enable')`?
```styl
@import hexo-config('back2top.sidebar') ? "back-to-top-sidebar" : "back-to-top" if hexo-config('back2top.enable');
```

**Comment by JiangTJ** [comment]
image
image
No scrollpercent....
see 
- https://gitlab.com/JiangTJ/next-test/blob/master/_config.yml
- https://jiangtj.gitlab.io/next-test/muse/

**Comment by stevenjoezhang** [comment]
Should work fine now..

**Comment by stevenjoezhang** [comment]
Ahh... I forgot something ... Updated again

**Comment by JiangTJ** [comment]
image
Breaking change... may need add describe in PR.

**Comment by ivan-nginx** [comment]
@stevenjoezhang see how to add readme when breaking changes exists.
And this is Feature, not Optimization. Because new option exists.

**Comment by stevenjoezhang** [comment]
@ivan-nginx I know, docs have already been added

**Comment by ivan-nginx** [comment]
@stevenjoezhang I mean – don't forget to set checkbox if docs already added.

**Comment by stevenjoezhang** [comment]
Sorry, I'll check again

File: layout/_partials/head/head.swig [hunk 2]
```diff
@@ -107,6 +107,8 @@
     scheme: '{{ theme.scheme }}',
     version: '{{ version }}',
     sidebar: {{ theme.sidebar | json_encode }},
+    back2top: {{ theme.back2top.enable }},
```
Comment (ivan-nginx) [inline layout/_partials/head/head.swig]: Need to refactor this under json, no need both `back2top` and `back2top_sidebar` parameters.

Comment (stevenjoezhang) [reply layout/_partials/head/head.swig]: Yes, it's a good idea. You can do it.

**Pull request merged**